### PR TITLE
Logic blocks

### PR DESCRIPTION
**Pull request by AnmAtAnm**

### CODE REVIEW

File: blocks/logic.js [hunk 1]
```diff
@@ -20,54 +20,290 @@
...
+    ],
+    "output": true,
+    "colour": "%{BKY_LOGIC_HUE}",
+    "tooltip": "%{BKY_LOGIC_TERNARY_TOOLTIP}",
+    "helpUrl": "%{BKY_LOGIC_TERNARY_HELPURL}",
+    "extensions": ["logic_ternary"]
+  }
+]);  // END JSON EXTRACT (Do not delete this comment.)
+
+Blockly.defineBlocksWithJsonArray([ // Mutator blocks. Do not extract.
```
Comment (rachel-fenichel) [inline blocks/logic.js]: "Do not extract"?

File: blocks/logic.js [hunk 2]
```diff
@@ -20,54 +20,290 @@
...
+  'NEQ': '%{BKY_LOGIC_COMPARE_TOOLTIP_NEQ}',
+  'LT': '%{BKY_LOGIC_COMPARE_TOOLTIP_LT}',
+  'LTE': '%{BKY_LOGIC_COMPARE_TOOLTIP_LTE}',
+  'GT': '%{BKY_LOGIC_COMPARE_TOOLTIP_GT}',
+  'GTE': '%{BKY_LOGIC_COMPARE_TOOLTIP_GTE}',
+
+  // logic_operation
+  'AND': '%{BKY_LOGIC_OPERATION_TOOLTIP_AND}',
+  'OR': '%{BKY_LOGIC_OPERATION_TOOLTIP_OR}'
+};
```
Comment (rachel-fenichel) [inline blocks/logic.js]: Nit: missing newline

File: blocks/logic.js [hunk 3]
```diff
@@ -20,54 +20,290 @@
...
+Blockly.Extensions.register("logic_op_tooltip",
+  Blockly.Extensions.buildTooltipForDropdown(
+    'OP', Blockly.Constants.Logic.LOGIC_OP_TOOLTIPS));
+
```
Comment (rachel-fenichel) [inline blocks/logic.js]: Nit: extra newline

File: blocks/logic.js [hunk 4]
```diff
@@ -378,115 +550,36 @@ Blockly.Blocks['logic_compare'] = {
...
-      ],
-      "output": "Boolean",
-      "colour": Blockly.Blocks.logic.HUE,
-      "tooltip": Blockly.Msg.LOGIC_NEGATE_TOOLTIP,
-      "helpUrl": Blockly.Msg.LOGIC_NEGATE_HELPURL
-    });
-  }
+  // Add onchange handler to ensure types are compatable.
+  this.mixin(Blockly.Constants.Logic.LOGIC_COMPARE_ONCHANGE_MIXIN);
 };
```
Comment (rachel-fenichel) [inline blocks/logic.js]: Nit: missing newline

File: blocks/logic.js [hunk 5]
```diff
@@ -518,3 +611,5 @@ Blockly.Blocks['logic_ternary'] = {
     this.prevParentConnection_ = parentConnection;
   }
 };
```
Comment (rachel-fenichel) [inline blocks/logic.js]: Nit: missing newline

File: blocks/logic.js (hunk 1 above)
Comment (AnmAtAnm) [inline blocks/logic.js]: Note the lack of "BEGIN JSON EXTRACT" is intentional.

Note that I've now added the beginning marker to distinguish it from sections like this set of mutator blocks (useless to Android and iOS).

File: blocks/logic.js (hunk 2 above)
Comment (AnmAtAnm) [inline blocks/logic.js]: Added such newlines everywhere.

**Pull request merged**